lib/rules/no-unsafe-finally.js

```diff
@@ -9,7 +9,8 @@
 // Helpers
 //------------------------------------------------------------------------------
 
-var SENTINEL_NODE_TYPE = /^(?:Program|(?:Function|Class)(?:Declaration|Expression)|ArrowFunctionExpression)$/;
+var SENTINEL_NODE_TYPE_RETURN_THROW = /^(?:Program)|(?:Function|Class)(?:Declaration|Expression)|ArrowFunctionExpression$/;
+var SENTINEL_NODE_TYPE_BREAK_CONTINUE = /^(?:Program)|(?:Function|Class)(?:Declaration|Expression)|ArrowFunctionExpression|DoWhileStatement|WhileStatement|ForStatement|SwitchStatement$/;
```

Comment:
@mysticatea oh, dang. You're right. May be we need a rule to warn on the "invalid" one then? :P